src/utils/guard_value.hpp

```diff
@@ -0,0 +1,35 @@
+/*
+   Copyright (C) 2003 - 2018 the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#pragma once
+
+namespace utils {
+
+template<typename T>
+class guard_value {
```

Comment:
Needs documentation. Please don't wait for a review comment on each class.

Reply:
I think this class is pretty self-explanatory, but sure, I'll add a comment.